src/NHSD.BuyingCatalogue.Organisations.Api/Controllers/OdsController.cs

```diff
@@ -36,6 +38,11 @@ public async Task<ActionResult> GetByOdsCodeAsync(string odsCode)
             if (!(odsOrganisation.IsActive && odsOrganisation.IsBuyerOrganisation))
                 return new StatusCodeResult(StatusCodes.Status406NotAcceptable);
 
+            var organisation = await organisationRepository.GetByOdsCodeAsync(odsCode);
+
+            if (organisation == null)
+                return NotFound();
```

Comment:
We can't return a 404 here as this endpoint is used by the admin application to lookup organization details when an authority/admin user wants to add an organization. However, at that point the organization won't exist in the DB so it will break the admin add organization journey.

Reply:
Updated